docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,503 @@
+      - any: true
+    principals:
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "spiffe://mesh-1/backend"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://version/v1"
+```
+
+In order request to be allowed, at least 1 policy should be matched. 
+
+#### Converting MeshTrafficPermission to Envoy NetworkRBAC
+
+1. Concatenate "from" arrays from all matched MeshTrafficPermissions 
+
+Input
+```yaml
+type: MeshTrafficPermission
+name: global
+spec:
+  targetRef:
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: Mesh
+      conf:
+        action: DENY
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          kuma.io/zone: us-east
+      conf:
+        action: ALLOW
+---
+type: MeshTrafficPermission
+name: backend
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  from:
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          env: dev
+      conf:
+        action: DENY
+```
+
+Concatenated "from" array
+```yaml
+from:
+  - targetRef:
+      kind: Mesh
+    conf:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    conf:
+      action: ALLOW
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        env: dev
+    conf:
+      action: DENY
+```
+
+2. Create a full rule-based view (with negations) 
+
+Regular [rule-based view](005-policy-matching.md#rule-based-view) doesn't work well for NetworkRBAC generation,
+because rule-based view relies on the order, but NetworkRBAC doesn't. 
+
+In this example, if a request matches "rule 2" this means it doesn't match "rule 1":
+```yaml
+rules:
+  - targetRef: # rule 1
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: dev
+    conf:
+      action: DENY
+  - targetRef: # rule 2
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    conf:
+      action: ALLOW
+```
+
+So "rule 2" implicitly matches requests from zone "us-east" that are **not** from env "dev".
+That's why it's important to have a full rule-based view with negations ("!" means "not"):
+
+```yaml
+rules:
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: dev
+    conf:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: !dev
+    conf:
+      action: ALLOW
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: !us-east
+        env: dev
+    conf:
+      action: DENY 
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: !us-east
+        env: !dev
+    conf:
+      action: DENY
+```
+
+See [how to generate a full rule-based view with negations](#how-to-generate-a-full-rule-based-view-with-negations)
```

Comment:
Do we agree that this "rule-based" view with negations is only an internal and conceptual thing used here to explain how we'll do the mapping, It will never be exposed to the user?

Reply:
Yes, we don't need to expose that to users